File: 08 - JAVA/01 - POO/Demo/Demo/src/main/java/org/example/DemoHeritage/Chien.java
package org.example.DemoHeritage;

public class Chien extends Canide{
    private String race;
    public Chien(String race,String nom, int age) {
        super(nom, age);
        this.race = race;
    }

    @Override
    public void crier(){
        System.out.println("le chien abboie !");
    }

    public void aboyer (){
        crier();
        System.out.println("aboie depuis aboyer");
    }

    @Override
    public String toString() {
        return "Chien{" +
                "race='" + race + '\'' +
                ", nom='" + nom + '\'' +
                ", age=" + age +
                '}';
    }
}

File: Thibaud_exo/Exo_JavaScript/Exo25/javaScript/class/chien.js
let id = 0;

export class Chien {

    constructor(nom, race, age) {
        this.id = ++id;
        this.nom = nom;
        this.race = race;
        this.age = age;
    }

    display() {
        return `Chien { Id ${this.id}, Nom ${this.nom}, Race ${this.race}, Age ${this.age} }`;
    }
}
